### PR TITLE
fix(table)!: Add widths parameter to new()

### PR DESCRIPTION
**Pull request by joshka**
- style(tabs): fix doc formatting
- feat(layout): Accept IntoIterator for constraints
- fix(table)!: Add widths parameter to new()

### CODE REVIEW

File: BREAKING-CHANGES.md [hunk 1]
```diff
@@ -38,7 +39,32 @@ This is a quick summary of the sections below:
 
 ## Unreleased (v0.24.1)
 
-### `Table::widths()` now accepts `AsRef<[Constraint]>` ([#628])
+### `Table::new()` now requires specifying the widths of the columrs (#664)
+
+
+Previously `Table`s could be constructed without widths. In almost all cases this is an error.
```
Comment (orhun) [inline BREAKING-CHANGES.md:45]: > In almost all cases this is an error.

I'm curious, did we get any bug reports about this + what kind of an errors users get if no widths are specified?

Comment (orhun) [reply BREAKING-CHANGES.md:45]: Great, that makes sense.

**Comment by joshka** [comment]
rebased on the other table changes

**Review by orhun — APPROVED** [review]

**Review by Valentin271 — APPROVED** [review]
LGTM!

**Pull request merged**